core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java

```diff
@@ -104,7 +104,7 @@ public ViewMetadata refresh() {
   @Override
   public void commit(ViewMetadata base, ViewMetadata metadata) {
     // if the metadata is already out of date, reject it
-    if (base != current()) {
+    if (!current().equals(base)) {
```

Comment:
the previous check is actually correct

Reply:
<img width="1518" alt="Screenshot 2024-08-08 at 6 49 29 PM" src="https://github.com/user-attachments/assets/f22f43c2-b6ce-4a59-8a19-0c7a488f065a">

I am seeing above warning with build. Is it ignorable ?

Reply:
yes, in this case this warning should be ignored

Reply:
@nk1506 are you going to add a suppresswarning for this case?